ibm/resource_ibm_is_virtual_endpoint_gateway_test.go

```diff
@@ -0,0 +1,223 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckisVirtualEndpointGatewayExists(name, &monitorOne),
+					testAccisVirtualEndpointGatewayManuallyDelete(&monitorOne),
+				),
+			},
+			{
+				Config: testAccCheckisVirtualEndpointGatewayConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckisVirtualEndpointGatewayExists(name, &monitorTwo),
+					func(state *terraform.State) error {
+						if monitorOne == monitorTwo {
+							return fmt.Errorf("load balancer monitor id is unchanged even after we thought we deleted it ( %s )",
+								monitorTwo)
+						}
+						return nil
+					},
+				),
+			},
+		},
+	})
+}
+
+func testAccisVirtualEndpointGatewayManuallyDelete(tfEndpointGwID *string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		sess, err := testAccProvider.Meta().(ClientSession).VpcV1API()
+		if err != nil {
+			return err
+		}
+		tfEndpointGw := *tfEndpointGwID
+		opt := sess.NewDeleteEndpointGatewayOptions(tfEndpointGw)
+		response, err := sess.DeleteEndpointGateway(opt)
+		if err != nil {
+			return fmt.Errorf("Delete Endpoint Gateway failed: %v", response)
+		}
+		return nil
+	}
+}
+
+func testAccCheckisVirtualEndpointGatewayDestroy(s *terraform.State) error {
+	sess, err := testAccProvider.Meta().(ClientSession).VpcV1API()
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_is_virtual_endpoint_gateway" {
+			continue
+		}
+		opt := sess.NewGetEndpointGatewayOptions(rs.Primary.ID)
+		_, response, err := sess.GetEndpointGateway(opt)
+		if err == nil {
+			return fmt.Errorf("Endpoint Gateway still exists: %v", response)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckisVirtualEndpointGatewayExists(n string, tfEndpointGwID *string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No endpoint gateway ID is set")
+		}
+
+		sess, err := testAccProvider.Meta().(ClientSession).VpcV1API()
+		if err != nil {
+			return err
+		}
+
+		opt := sess.NewGetEndpointGatewayOptions(rs.Primary.ID)
+		result, response, err := sess.GetEndpointGateway(opt)
+		if err != nil {
+			return fmt.Errorf("Endpoint Gateway does not exist: %s", response)
+		}
+		*tfEndpointGwID = *result.ID
+		return nil
+	}
+}
+
+func testAccCheckisVirtualEndpointGatewayConfigBasic() string {
+	return fmt.Sprintf(`
+	data "ibm_resource_group" "test_acc" {
+		name = "default"
+    }
+	resource "ibm_is_vpc" "testacc_vpc" {
+		name = "test-vpe-network-vpc"
+		resource_group = data.ibm_resource_group.test_acc.id
+	}
+	resource "ibm_is_subnet" "testacc_subnet" {
+		name = "test-vpe-subnet1"
+		vpc = ibm_is_vpc.testacc_vpc.id
+		zone = "%[1]s"
+		ipv4_cidr_block = "%[2]s"
+		resource_group = data.ibm_resource_group.test_acc.id
+	}
+	resource "ibm_is_virtual_endpoint_gateway" "endpoint_gateway" {
```

Comment:
Don't hardcode the names of the resouces test-vpe-subnet1,test-vpe-network-vpc...like this in all testcases
See the remaining VPC testcases and use the same

Reply:
handled